### PR TITLE
Feature/tilelistdownload3

### PR DESCRIPTION
**Pull request by wibge**
I had trouble figuring out how to communicate the list of tiles between the MapBox library, and the application. I made a MGLTileID class, but maybe there's a better way.

### CODE REVIEW

File: platform/ios/app/MBXOfflinePacksTableViewController.m [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #import "MBXOfflinePacksTableViewController.h"
 
+#import "MGLTileID.h"
```
Comment (wibge) [inline platform/ios/app/MBXOfflinePacksTableViewController.m]: how do i get this to import with Mapbox/MapBox.h

File: platform/darwin/src/MGLTilePyramidOfflineRegion.mm [hunk 2]
```diff
@@ -49,14 +58,24 @@ - (instancetype)initWithStyleURL:(NSURL *)styleURL bounds:(MGLCoordinateBounds)b
...
 
 - (instancetype)initWithOfflineRegionDefinition:(const mbgl::OfflineRegionDefinition &)definition {
     NSURL *styleURL = [NSURL URLWithString:@(definition.styleURL.c_str())];
     MGLCoordinateBounds bounds = MGLCoordinateBoundsFromLatLngBounds(definition.bounds);
-    return [self initWithStyleURL:styleURL bounds:bounds fromZoomLevel:definition.minZoom toZoomLevel:definition.maxZoom];
+    return [self initWithStyleURL:styleURL bounds:bounds fromZoomLevel:definition.minZoom toZoomLevel:definition.maxZoom tileList:nil];
+}
+
+// TODO: why can't it find this function in MGLTileID
+MGLTileID MGLTileIDFromKey(uint64_t tilekey) {
```
Comment (wibge) [inline platform/darwin/src/MGLTilePyramidOfflineRegion.mm]: compiler couldn't find this function in MGLTileID. I don't know why.

File: platform/darwin/src/MGLTileID.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by Anna Johnson on 2/7/17.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+typedef struct
+{
+  uint32_t x, y, z;
```
Comment (JesseCrocker) [inline platform/darwin/src/MGLTileID.h]: For consistency with CanonicalTileID this should be uint_8t

File: platform/darwin/src/MGLTilePyramidOfflineRegion.h [hunk 4]
```diff
@@ -44,6 +44,11 @@ MGL_EXPORT
  */
 @property (nonatomic, readonly) double maximumZoomLevel;
 
+/**
+ A specific list of tileIds to download for this region
+*/
+@property (nonatomic, readonly) NSArray* tileList;
```
Comment (JesseCrocker) [inline platform/darwin/src/MGLTilePyramidOfflineRegion.h]: this should be nullable

File: platform/darwin/src/MGLTilePyramidOfflineRegion.mm [hunk 5]
```diff
@@ -65,10 +84,17 @@ - (instancetype)initWithOfflineRegionDefinition:(const mbgl::OfflineRegionDefini
 #elif TARGET_OS_MAC
     const float scaleFactor = [NSScreen mainScreen].backingScaleFactor;
 #endif
+    
+    std::vector<mbgl::CanonicalTileID> tileVector;
+    for (NSNumber* value in _tileList) {
```
Comment (JesseCrocker) [inline platform/darwin/src/MGLTilePyramidOfflineRegion.mm]: tileVector.reserve(_tileList.count)

File: platform/ios/app/MBXOfflinePacksTableViewController.m (hunk 1 above)
Comment (JesseCrocker) [reply platform/ios/app/MBXOfflinePacksTableViewController.m]: Put it in mapbox.h

**Review by JesseCrocker — CHANGES_REQUESTED** [review]

**Pull request closed without merging**